### PR TITLE
[HOPSWORKS-908] Airflow as a service

### PR DESCRIPTION
**Pull request by kouzant**
https://logicalclocks.atlassian.net/browse/HOPSWORKS-908

Check JIRA for dependencies

### CODE REVIEW

File: recipes/install.rb [hunk 1]
```diff
@@ -313,6 +314,26 @@
               }
             }
           },
+          'airflowPool' => {
```
Comment (SirOibaf) [inline recipes/install.rb]: If I remember correctly, during an update this part is not executed.

File: templates/default/convert-ipython-notebook.sh.erb [hunk 2]
```diff
@@ -54,6 +54,14 @@ if [ $? -ne 0 ] ; then
...
 fi    
 
+
+echo "
+from pyspark import SparkContext
+sc = SparkContext(\"yarn\", \"${base}\")
+
+" > /tmp/"${base}".py
+cat "${base}".py >> /tmp/"${base}".py
+
```
Comment (SirOibaf) [inline templates/default/convert-ipython-notebook.sh.erb]: According to @robzor92  this not correct. The PySpark kernel creates a `SparkSession` object named `spark` not a `SparkContext` named sc.

Comment (robzor92) [reply templates/default/convert-ipython-notebook.sh.erb]: I am not opposed the idea. But as it is right now it is wrong if we want to make it compatible with Jupyter. It is as Fabio said. We should not get the **SparkContext**, but the **SparkSession**. And it **must** be named **spark**. You get the SparkSession like this. https://github.com/logicalclocks/hops-util-py/blob/7df48d2ecc6153513f6aed097335a16fa49b4ca9/hops/util.py#L408

Comment (robzor92) [reply templates/default/convert-ipython-notebook.sh.erb]: I am also assuming this is not tested. So unless it is done so properly after the changes, please remove this code. I will reintroduce it in HOPSWORKS-716 in that case.

Comment (kouzant) [reply templates/default/convert-ipython-notebook.sh.erb]: I removed this part of code

File: recipes/install.rb (hunk 1 above)
Comment (kouzant) [reply recipes/install.rb]: Code is there but since I can't test it on time, it will be merged on another commit.

**Pull request merged**